src/network/tcpsocket.c

```diff
@@ -228,16 +225,19 @@ void send_tcp(char *msg) {
     print_tcp_array();
     if (network_config.use_symm_enc) {
         int length_enc;
-        size_t msglen = strlen(msg);
-        char *enc = gcrypt_encrypt_msg(msg, msglen + 1, &length_enc);
+        size_t msglen = strlen(msg)+1;
+        char *enc = gcrypt_encrypt_msg(msg, msglen, &length_enc);
 
-        char *base64_enc_str = malloc(B64_ENCODE_LEN(length_enc));
-        size_t base64_enc_length = b64_encode(enc, length_enc, base64_enc_str, B64_ENCODE_LEN(length_enc));
         struct network_con_s *con;
+        size_t final_len = length_enc + sizeof(size_t);
+        char *final_str = malloc(final_len);
+        size_t *msg_len = (size_t*)final_str;
+        *msg_len = final_len;
```

Comment:
WTF happens on the above two lines?

Reply:
to leave some space for msg length including header.

since final_str or most msg is a pointer to char array, I have to convert char pointer to size_t pointer.

Reply:
max number could fit in one byte or char is 255, not big enough.

I set it to size_t ( 4 byte integer)